godoc: add page

danieldc opened the pull request:
----
<!-- Thank you for sending a PR! -->
<!-- Please perform the following checks and check all the boxes that apply. -->
<!-- If your PR does not create a command page,
     you can remove the first two checklist items. -->
<!-- If your PR neither creates nor edits a command page (e.g. README edits, etc.)
     you can simply remove the entire checklist. -->

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

agnivade commented on pages/common/godoc.md:
```diff
@@ -0,0 +1,11 @@
+
+- Serve documentation as an HTML web server on http://localhost:6060/":
+
+`godoc -http=:6060`
```
6060 in tokens (`{{}}`)

pxgamer commented on pages/common/godoc.md:
```diff
@@ -0,0 +1,11 @@
+
+`godoc {{fmt}} {{Printf}}`
+
+- Serve documentation as an HTML web server on http://localhost:6060/":
```
Yeah, definitely a bit more readable. 👍 Also, as there wasn't a starting quote on the URL, this looks a bit odd.

danieldc replied:
done.

danieldc commented:
Added `godoc {{package}}`, simplified wording for web serving use case and `godoc {{package}} {{name}}`.

(sorry for delay -- I'm traveling and wifi coverage is spotty).

danieldc commented on pages/common/godoc.md:
```diff
@@ -0,0 +1,11 @@
+
+> Show documentation for go packages.
+
+- Display help for name (named "Printf" in this case) of package (named "fmt" in this case):
```
done.

agnivade commented:
> About index related commands: would you be OK accepting the PR without them?

Sure, I can add them myself. Or are you suggesting you don't want the index commands in the page ? I think it is a very useful option to anyone seriously using godoc.

danieldc commented:
>  Sure, I can add them myself. Or are you suggesting you don't want the index commands in the page 

I haven't interacted with enough go users to judge this, and I don't use them myself hence I don't feel that I'm the right person to do them.
If you think they are useful, then I think it will be great to add them.

FWIW, here's what I would have done (side note: I don't mind if you add this to this PR, change the content as you see fit, or do it in a different PR)

> - Serve documentation as a web server on port "6060", and enable word searching (i.e. "full text search"), rather than just indentifiers:
> 
> `godoc -http=:{{6060}} -indexi`
> 
> - Serve documentation as a web server on port "6060", and use the given file as the full text search index. The second command will refresh the index.
> `godoc -http=:{{6060}} -index -index_files=/tmp/godoc.fsix`
> `godoc -goroot=/usr/local/go -index -index_files=/tmp/godoc.fsix -write_index`

agnivade commented:
Thank you. I will open a separate PR. Since we have space left for other commands, it should be fine to add these.
